src/DatePicker/Calendar.js

```diff
@@ -81,14 +83,19 @@ class Calendar extends Component {
         selectedDate: date,
       });
     }
+    if (nextProps.displayMode !== this.props.displayMode) {
```

Comment:
That block is making the `displayMode` implementation an hybride between a controlled and an uncontrolled property.
I don't that this hybride state is good patern. E.g the initial value is A, this value is mutated to B by a setState. How can we use the property to go back to the A state? It can't be done directly, we need to move to state C so then, you can go to state A.